resources/dw/resource_hive_vw_test.go

```diff
@@ -0,0 +1,159 @@
+//
+// This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, either express or implied. Refer to the License for the specific
+// permissions and limitations governing your use of the file.
+
+package dw
+
+import (
+	"context"
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/cdp"
+	dwclient "github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/gen/dw/client"
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/gen/dw/client/operations"
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/gen/dw/models"
+	mocks "github.com/cloudera/terraform-provider-cdp/mocks/github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/gen/dw/client/operations"
+	"github.com/go-openapi/runtime"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/suite"
+	"testing"
+)
+
+type MockTransport struct {
+	runtime.ClientTransport
+}
+
+func NewDwApi(client *mocks.MockDwClientService) *hiveResource {
+	return &hiveResource{
+		client: &cdp.Client{
+			Dw: &dwclient.Dw{
+				Operations: client,
+				Transport:  MockTransport{},
+			}}}
+}
+
+func createRawHiveResource() tftypes.Value {
+	return tftypes.NewValue(
+		tftypes.Object{
+			AttributeTypes: map[string]tftypes.Type{
+				"id":                  tftypes.String,
+				"cluster_id":          tftypes.String,
+				"database_catalog_id": tftypes.String,
+				"name":                tftypes.String,
+			}},
+		map[string]tftypes.Value{
+			"id":                  tftypes.NewValue(tftypes.String, ""),
+			"cluster_id":          tftypes.NewValue(tftypes.String, "cluster-id"),
+			"database_catalog_id": tftypes.NewValue(tftypes.String, "database-catalog-id"),
+			"name":                tftypes.NewValue(tftypes.String, ""),
+		})
+}
+
+type HiveTestSuite struct {
+	suite.Suite
+	expectedCreateResponse *operations.CreateVwOK
+}
+
+func TestHive(t *testing.T) {
+	suite.Run(t, new(HiveTestSuite))
+}
+
+func (suite *HiveTestSuite) SetupTest() {
+	suite.expectedCreateResponse = &operations.CreateVwOK{Payload: &models.CreateVwResponse{
+		VwID: "test-id",
+	}}
+
+}
+
+func (suite *HiveTestSuite) TestHiveMetadata() {
+	ctx := context.TODO()
+	client := new(mocks.MockDwClientService)
+	dwApi := NewDwApi(client)
+
+	req := resource.MetadataRequest{ProviderTypeName: "dw"}
+	resp := &resource.MetadataResponse{}
+
+	// Function under test
+	dwApi.Metadata(ctx, req, resp)
+	suite.Equal("dw_vw_hive", resp.TypeName)
+}
+
+func (suite *HiveTestSuite) TestHiveSchema() {
+	ctx := context.TODO()
+	client := new(mocks.MockDwClientService)
+	dwApi := NewDwApi(client)
+
+	req := resource.SchemaRequest{}
+	resp := &resource.SchemaResponse{}
+
+	// Function under test
+	dwApi.Schema(ctx, req, resp)
+	suite.Equal(hiveSchema, resp.Schema)
```

Comment:
Never use production code in tests for comparison... in this test if you change the `hiveSchema` accidentally your test will still PASS too w/o noticing the unexpected change.

Reply:
I've added a testHiveSchema, this is what you meant? This is good for schema testing, but for creation test it is okay to use the production hiveSchema?

Reply:
yes sure